### PR TITLE
[posttest] Add collect_support with posttest marker for collecting logs and dumps 

### PR DESCRIPTION
**Pull request by bingwang-ms**
### Description of PR

Summary:
Add collect_support with posttest marker for collecting logs and dumps.
1. Add new script test_posttest.py for jobs after tests;
2. Rename test_cleanup_testbed.py to test_pretest.py, because there might be another job needed to be done before running tests. It's more clear and reasonable to write new cases in test_pretest.py than creating new script file.

### Type of change

- [ ] Bug fix
- [x] Testbed and Framework(new/improvement)
- [ ] Test case(new/improvement)

#### What is the motivation for this PR?
This PR add a util for collecting techsupport after tests.
Since nightly test on Jenkins will do a cleanup at the beginning of tests, we need a method to save history logs and dumps. This util does the job.
#### How did you do it?
Add a new case with posttest marker. The run_tests.sh script will call it after running all tests. The dump file will be saved to tests/logs.
#### How did you verify/test it?
Verified on Arista-7260cx3, and confirmed dump file is saved in sonic-mgmt container.
```
py.test --inventory ../ansible/str,../ansible/veos --host-pattern str-7260cx3-acs-1 --module-path ../ansible/library/ --testbed vms7-t0-7260-1 --testbed_file ../ansible/testbed.csv --junit-xml=tr.xml --log-cli-level warn --collect_techsupport=False --topology=t0,any,util -m posttest
========================================================================================= test session starts =========================================================================================
platform linux2 -- Python 2.7.12, pytest-4.6.5, py-1.8.1, pluggy-0.13.1 -- /usr/bin/python
cachedir: .pytest_cache
ansible: 2.8.7
rootdir: /data/Networking-acs-sonic-mgmt
plugins: forked-1.1.3, xdist-1.28.0, ansible-2.2.2, repeat-0.8.0
collected 1056 items / 1055 deselected / 1 selected                                                                                                                                                   

test_posttest.py::test_collect_techsupport PASSED                                                                                                                                               [100%]
------------------------------------------------------------------ generated xml file: /data/Networking-acs-sonic-mgmt/tests/tr.xml -------------------------------------------------------------------
======================================================================= 1 passed, 1055 deselected, 11 warnings in 60.29 seconds =======================================================================

```
#### Any platform specific information?
No.
#### Supported testbed topology if it's a new test case?
N/A
### Documentation 

No.

### CODE REVIEW

File: tests/test_posttest.py [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+    we need a method to save history logs and dumps. This util does the job.
+    """
+    logger.info("Collecting techsupport since yesterday")
+    # Because Jenkins is configured to save artifacts from tests/logs,
+    # and this util is mainly designed for running on Jenkins,
+    # save path is fixed to logs for now.
+    TECHSUPPORT_SAVE_PATH = 'logs/'
+    out = duthost.command("show techsupport --since yesterday", module_ignore_errors=True)
+    if out['rc'] == 0:
+        tar_file = [j for j in out['stdout'].split('\n') if j != ''][-1]
```
Comment (wangxin) [inline tests/test_posttest.py:27]: The command result should have a 'stdout_lines' field. You can directly use `out['stdout_lines']` here.

File: tests/test_posttest.py [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+
+    Since nightly test on Jenkins will do a cleanup at the beginning of tests,
+    we need a method to save history logs and dumps. This util does the job.
+    """
+    logger.info("Collecting techsupport since yesterday")
+    # Because Jenkins is configured to save artifacts from tests/logs,
+    # and this util is mainly designed for running on Jenkins,
+    # save path is fixed to logs for now.
+    TECHSUPPORT_SAVE_PATH = 'logs/'
+    out = duthost.command("show techsupport --since yesterday", module_ignore_errors=True)
```
Comment (wangxin) [inline tests/test_posttest.py:25]: You can consider running `generate_dump -s yesterday` command which is less verbose. The `show techsupport` always calls `generate_dump` with verbose on under the hood: https://github.com/Azure/sonic-utilities/blob/master/show/main.py#L1201

File: tests/test_posttest.py (hunk 1 above)
Comment (bingwang-ms) [reply tests/test_posttest.py]: Thanks Xin. Updated.

File: tests/test_posttest.py (hunk 2 above)
Comment (bingwang-ms) [reply tests/test_posttest.py]: Thanks Xin. Updated.

File: tests/test_posttest.py [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+    # Because Jenkins is configured to save artifacts from tests/logs,
+    # and this util is mainly designed for running on Jenkins,
+    # save path is fixed to logs for now.
+    TECHSUPPORT_SAVE_PATH = 'logs/'
+    out = duthost.command("generate_dump -s yesterday", module_ignore_errors=True)
+    if out['rc'] == 0:
+        tar_file = out['stdout_lines'][-1]
+        duthost.fetch(src=tar_file, dest=TECHSUPPORT_SAVE_PATH, flat=True)
+
+    assert True
```
Comment (wangxin) [inline tests/test_posttest.py:30]: Is this `assert True` required?

Comment (bingwang-ms) [reply tests/test_posttest.py:30]: Not really. But I would like to keep it because I want to indicate that this case always passes explicitly.

**Review by wangxin — APPROVED** [review]

**Pull request merged**